File: peertube_client.py
import requests
import sys

# 5 trending videos on peertube.social
# python peertube_client.py https://peertube.tv trending 5

def fetch_peertube_videos(instance_url, scope='trending', limit=5):
    """
    scope: one of 'local', 'trending', 'subscriptions' (subscriptions needs auth), 
           'recommended', etc.—but 'trending' and 'local' work without login.
    """
    base = instance_url.rstrip('/')
    url = f"{base}/api/v1/videos"
    params = {
        'scope': scope,
        'limit': limit
    }
    resp = requests.get(url, params=params)
    resp.raise_for_status()
    return resp.json()

def main():
    if len(sys.argv) < 2:
        print("Usage: python peertube_client.py <instance_url> [scope] [limit]")
        print("  scope: local or trending (default: trending)")
        sys.exit(1)

    instance = sys.argv[1]             # e.g. https://peertube.social
    scope    = sys.argv[2] if len(sys.argv) > 2 else 'trending'
    limit    = int(sys.argv[3]) if len(sys.argv) > 3 else 5

    videos = fetch_peertube_videos(instance, scope, limit)
    for vid in videos.get('data', videos):
        title = vid.get('name', vid.get('title', 'untitled'))
        link  = vid.get('videoUrl') or vid.get('url')
        print(f"{title}\n{link}\n")

if __name__ == '__main__':
    main()

File: node/public/clients/peertube_client.js
const axios = require('axios');

//US: node peertube_client.js <instance_url> [scope] [limit]

/**
 * Fetch the latest videos from a PeerTube instance.
 * @param {string} instanceUrl - Base URL of the PeerTube instance (e.g. https://peertube.social)
 * @param {string} scope - 'local' or 'trending' (default: 'trending')
 * @param {number} limit - Number of videos to fetch (default: 5)
 * @returns {Promise<Array>} - List of video objects
 */
async function fetchPeerTubeVideos(instanceUrl, scope = 'trending', limit = 5) {
  const base = instanceUrl.replace(/\/+$/, '');
  const url = `${base}/api/v1/videos`;
  try {
    const response = await axios.get(url, {
      params: {
        scope,
        limit
      }
    });
    return response.data.data || response.data;
  } catch (error) {
    console.error('Error fetching videos:', error.response ? error.response.statusText : error.message);
    if (error.response) {
      console.error('Status code:', error.response.status);
      console.error('Response:', error.response.data);
    }
    return [];
  }
}

// CLI entry point
// if (require.main === module) {
//   const [,, instance, cliScope, cliLimit] = process.argv;
//   if (!instance) {
//     console.error('Usage: node peertube_client.js <instance_url> [scope] [limit]');
//     console.error("  scope: 'local' or 'trending' (default: 'trending')");
//     process.exit(1);
//   }
//   const scope = cliScope || 'trending';
//   const limit = parseInt(cliLimit, 10) || 5;

//   (async () => {
//     const videos = await fetchPeerTubeVideos(instance, scope, limit);
//     videos.forEach(vid => {
//       const title = vid.name || vid.title || 'untitled';
//       const link = vid.videoUrl || vid.url || '';
//       console.log(`${title}\n${link}\n`);
//     });
//   })();
// }
